_posts/2022-01-23-feature-deep-dive-opensearch-sql-basic-queries.markdown

```diff
@@ -0,0 +1,53 @@
+
+## How to use OpenSearch SQL?
+
+For OpenSearch user, you could use the familiar REST interface `_plugin/sql` with your query in query body. 
+For DBMS user, you could use [JDBC](https://github.com/opensearch-project/sql/tree/main/sql-jdbc)/[ODBC](https://github.com/opensearch-project/sql/tree/main/sql-odbc) driver to connect OpenSearch domain. You could download JDBC/ODBC driver. - add download link
+For Dashboards user, you could use [OpenSearch Dashboard Query Workbench](https://opensearch.org/docs/latest/search-plugins/sql/workbench/) to easily run on-demand SQL queries and download results
+An additional OpenSearch SQL CLI tool is provided for interactive SQL execution.
+
+## Some Examples
+
+### Basic Queries
+
+*“Find error logs where response code is 404 or 503”.* 
+You could run ODQL in Query Workbench. The result is in tabular format. You could download result in JSON, JDBC, CSV and Text format. Find more examples in [here](https://github.com/opensearch-project/sql/blob/main/docs/user/interfaces/protocol.rst).
+![basic queries](/assets/media/blog-images/2022-01-30-opensearch-sql-basic-queries/basic_queries.gif){: .img-fluid}
+### Functions
+
+*“Find error logs where response code is 404 or 503, How many distinct host”* 
+You could run ODQL in Dashboards notebook also. There are 4 distinct hosts for about query,  result is in table format. More [aggregations](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/aggregations.rst) and [functions](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst) are supported.
+![functions](/assets/media/blog-images/2022-01-30-opensearch-sql-basic-queries/functions.gif){: .img-fluid}
+### Explain
+
+*“How does ODQL execute in OpenSearch”*. Let’s explain an example query. OpenSearch SQL explain endpoint returns the query execution plan. For example, the above query two operators.
+
+* ProjectOperator, execute in coordinate node, read output from child operator and project originCountry and count fields.
+* FilterOperator, execute in coordinator node, read output from child operator and filter docs which count > 500.
+* OpenSearchIndexScan, it is a **DSL query** which will be executed in OpenSearch through search endpoint**.**
+
+![explain](/assets/media/blog-images/2022-01-30-opensearch-sql-basic-queries/explain.gif){: .img-fluid}
+## Inside SQL Engines
+
+Internally, query will go through five major components in query engine. (1) **Language Processor** parse the query string by following the grammar and generate the AST (Abstract Syntax Tree). (2) **Core Engine** analyze, optimize the AST and build the Logical Plan. (3) **Storage Engine** is an pluggable component which provide the catalog schema and storage specified optimization and implementation. (4) **Execution Engine** schedule and execute the physical plan. (5) **Protocol** parse the request and control the response format. 
```

Comment:
- Internally, query => Internally, a query
- in query engine => in the query engine
- parse the query => parses the query
- generate => generates
- "analyze, optimize the AST and build" => "analyzes and optimizes the AST and builds
- "is an pluggable component which provide" => "is a pluggable component which provides"
- "schedule and execute the physical plan" => "schedules and executes the physical plan"
- "parse the request and control" => "parses the request and controls"

Reply:
Done. Thanks a lot.